src/Visitor/IgnoreNode/IgnoreCodeCoverageAnnotation.php

```diff
@@ -33,34 +33,24 @@
 
 declare(strict_types=1);
 
-namespace Infection\Visitor;
+namespace Infection\Visitor\IgnoreNode;
 
 use PhpParser\Node;
 use PhpParser\Node\Stmt;
-use PhpParser\NodeTraverser;
-use PhpParser\NodeVisitorAbstract;
 
 /**
  * @internal
  */
-final class PhpUnitClassCodeCoverageIgnoreVisitor extends NodeVisitorAbstract
+final class IgnoreCodeCoverageAnnotation implements IgnoresNode
 {
-    public function enterNode(Node $node)
+    public function ignores(Node $node): bool
     {
-        if (!$node instanceof Stmt\ClassLike) {
-            return null;
+        if (!$node instanceof Stmt\ClassLike && !$node instanceof Stmt\ClassMethod) {
```

Comment:
I've opted to combine the two visitors into one here, as they were almost identical anyways.